### PR TITLE
Use live data in stats use cases

### PR DESCRIPTION
**Pull request by planarvoid**
In this PR I'm moving the LiveData sources down to the UseCases. I've introduced a shared interface for all the UseCases which also allows us to clear the value when the ViewModel is torn down.

I've moved all the use cases into a package to increase the readibility.

I've also fixed all the tests. The only thing missing now are the tests for the DB loading of the data. 

To test:
- Open stats
- All the stats block are loaded
- Refreshing the list works
- Scrolling around works
- Change the current site
- Go back to Stats
- You don't see the old stats but only an empty list that gets filled once the data is loaded

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/InsightsAllTimeUseCase.kt [hunk 1]
```diff
@@ -1,29 +1,41 @@
...
-            val response = insightsStore.fetchAllTimeInsights(site, forced)
-            val model = response.model
-            val error = response.error
+@Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    private val insightsStore: InsightsStore
+) : BaseInsightsUseCase(ALL_TIME_STATS, mainDispatcher) {
+    override suspend fun loadCachedData(site: SiteModel): InsightsItem? {
+        val dbModel = insightsStore.getAllTimeInsights(site)
+        return dbModel?.let { loadAllTimeInsightsItem(it) }
```
Comment (0nko) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/InsightsAllTimeUseCase.kt]: `loadAllTimeInsightsItem()` handles a `null` model by displaying a "no data" message. Shouldn't `loadCachedData` always return a non-nullable `InsightsItem` this way?

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/LatestPostSummaryUseCase.kt [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
     private val insightsStore: InsightsStore,
     private val latestPostSummaryMapper: LatestPostSummaryMapper
-) {
-    private val mutableNavigationTarget = MutableLiveData<NavigationTarget>()
-    val navigationTarget: LiveData<NavigationTarget> = mutableNavigationTarget
+) : BaseInsightsUseCase(LATEST_POST_SUMMARY, mainDispatcher) {
+    override suspend fun loadCachedData(site: SiteModel): InsightsItem? {
+        val dbModel = insightsStore.getLatestPostInsights(site)
+        return dbModel?.let { loadLatestPostSummaryItem(it) }
```
Comment (0nko) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/LatestPostSummaryUseCase.kt]: The same as `InsightsAllTimeUseCase`.

**Review by 0nko — COMMENTED** [review]
Looks cool! I just left a few comments in the code.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/InsightsAllTimeUseCase.kt (hunk 1 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/InsightsAllTimeUseCase.kt]: I think I'll try to make this consistent. If the model is null, I'll hide the block, if the data in the model is null, I'll show a block which has the "No data yet" message. The first use case should be logged as an error.

**Comment by planarvoid** [comment]
@0nko I've implemented the changes and I think it's ready for another check.

**Comment by 0nko** [comment]
The loading blocks work but the behavior is a little bit funky. You can see a bunch of blocks with the progress bar but you don't know what's actually loading. What do you think about including a block label at the top?

Another thing is that as the blocks are loaded and their height expands, the current position keeps getting pushed further down and at the end you end up in the middle of the list. Is there some way to keep the position at the top?

**Comment by planarvoid** [comment]
@0nko The loading blocks will have a different design. I'll look into keeping the position. Maybe we can jump to the top of the list once the list is completely loaded for the first time.

**Comment by planarvoid** [comment]
@0nko I've fixed the random position changes on Items loading. You just need to keep the Layout manager and dispatch the state after the load.

**Comment by 0nko** [comment]
Looks great! :shipit:

**Pull request merged**